src/DynamoCoreWpf/ViewModels/Search/NodeSearchElementViewModel.cs

```diff
@@ -306,18 +307,37 @@ private void AutoLayoutNodes(object sender, EventArgs e)
             var nodeView = (NodeView) sender;
             var dynamoViewModel = nodeView.ViewModel.DynamoViewModel;
 
+            nodeView.RepositionNodeHandler += OnRepositionNode;
             dynamoViewModel.CurrentSpace.DoGraphAutoLayout(true);
 
             DynamoSelection.Instance.ClearSelection();
 
+            
+        }
+
+        private void OnRepositionNode(object sender, EventArgs a)
+        {
+            var n = sender as NodeView;
+            var dynamoViewModel = n.ViewModel.DynamoViewModel;
+
+            DynamoSelection.Instance.Selection.Add(n.ViewModel.NodeModel);
+
+            n.ViewModel.WorkspaceViewModel.BeginDragSelectionWithUndoRecorder(new Point2D(n.ViewModel.X, n.ViewModel.Y));
+
+            dynamoViewModel.ExecuteCommand(new DynamoModel.DragSelectionCommand(
+                new Point2D(n.ViewModel.X - n.ActualWidth / 2.0, n.ViewModel.Y),
+                DynamoModel.DragSelectionCommand.Operation.EndDrag));
+
```

Comment:
Can you explain what you are doing here?

Reply:
By reading, this seems simply moving the newly placed node half the node width to the left to make sure it does not cover. You can see it's triggered when `OnSizeChanged` so the actual width is correct for the new node.  I am curious about the effect on other cases because this should also impact other regular cases. And also output node case is not covered here

Reply:
@QilongTang your understanding is correct. I'm simply moving the node to the left by half its width when its correct width is known, which happens on the `OnSizeChanged` event. However, there is still an issue here as this event doesn't get triggered for all nodes consistently. As per my observation, I think it happens for only NodeModel (UI) nodes. For regular nodes, the `ActualWidth` obtained in `NodeViewReady` is the real width of the node. 

So the issue is if the `OnSizeChanged` event is not triggered consistently for all nodes, the `RepositionNode` method will not be called in all cases, and we cannot unsubscribe events or close the undo action group consistently, which is a problem. I'm still trying to figure that out.